src/bridge.py

```diff
@@ -1,184 +1,87 @@
-    pubt2.publish(converted_w[1])
-    pubt3.publish(converted_w[2])
-    pubt4.publish(converted_w[3])
-    pubt5.publish(converted_w[4])
-    pubt6.publish(converted_w[5])
-    pubt7.publish(converted_w[6])
-    pubt8.publish(converted_w[7])
-
-def callback_pose(data):
-    global euler
-    
+def cb_thrusters(data):
+    # TODO - this needs to be N/N*m not us - publish from propulsion
+    pub_t1.publish(data.SURGE_PORT)
+    pub_t2.publish(data.SURGE_STAR)
+    pub_t3.publish(data.SWAY_BOW)
+    pub_t4.publish(data.SWAY_STERN)
+    pub_t5.publish(data.HEAVE_BOW_PORT)
+    pub_t6.publish(data.HEAVE_BOW_STAR)
+    pub_t7.publish(data.HEAVE_STERN_STAR)
+    pub_t8.publish(data.HEAVE_STERN_PORT)
+
+
+def cb_sim_pose(data):
     clarke_poses = data.poses[0]
     clarke_position = clarke_poses.position
     clarke_orientation = clarke_poses.orientation
-    pub_state_x.publish(clarke_position.x + 3)
-    pub_state_y.publish(clarke_position.y)
-    pub_state_z.publish(clarke_position.z)
-    
-    pose = Pose()
-    pose.position.x = clarke_position.x + 3
-    pose.position.y = clarke_position.y
-    pose.position.z = clarke_position.z
-    pose.orientation = clarke_orientation
-    pub_pose.publish(pose)
-
-    q = np.array([clarke_orientation.x, clarke_orientation.y, clarke_orientation.z, clarke_orientation.w])
-    conversion = transformations.euler_from_quaternion(q, 'rxyz')
-    theta_x = conversion[0]
-    theta_y = conversion[1]
-    theta_z = conversion[2]
-
-    angles = np.array([theta_x, theta_y, theta_z])*DEG_PER_RAD
-
-    for i in range(3):
-            if angles[i] - euler[i] > ANGLE_CHANGE_TOL:
-                euler[i] = angles[i] - 360
-            elif euler[i] - angles[i] > ANGLE_CHANGE_TOL:
-                euler[i] = angles[i] + 360
-            else:
-                euler[i] = angles[i]
-
-    pub_state_theta_x.publish(euler[0])
-    pub_state_theta_y.publish(euler[1])
-    pub_state_theta_z.publish(euler[2])
     
-def callback_imu(data):
+    dr_msg = DeadReckonReport()
+    dr_msg.x = clarke_position.x + 3 # TODO - sketch
+    dr_msg.y = clarke_position.y
+    dr_msg.z = clarke_position.z
+
+    depth = clarke_position.z # TODO - check with reference to which datum
+
+    # TODO - specify roll/pitch/yaw for dvl
+    pub_dvl_deadreckon.publish(dr_msg)
+    pub_depth_sensor.publish(depth)
+
+
+def cb_sim_imu(data):
+    # TODO - attach imu link to preserve orientation
     gyro = Vector3(-data.angular_velocity.z, -data.angular_velocity.y, -data.angular_velocity.x)
-    angular_velocity_pub.publish(gyro)
+    pub_imu_angular_vel.publish(gyro)
     
-    imu_ekf_quat_msg = SbgEkfQuat()
-    quat = Quaternion(-data.orientation.z, -data.orientation.y, -data.orientation.x, data.orientation.w)
-    imu_ekf_quat_msg.quaternion = quat
-    imu_quat_pub.publish(imu_ekf_quat_msg)
+    sbg_quat = SbgEkfQuat()
+    quat = Quaternion(-data.orientation.z, -data.orientation.y, -data.orientation.x, data.orientation.w) # TODO - incorrect?
```

Comment:
what's incorrect? The sim has a different coordinate system, so we had to manually change it before publishing

Reply:
Like it seems to work now, but I'm confused why that is, or what's going on. The quaternion q2 = <-x, -y, -z, w> would represent the opposite rotation direction (about the same axis) as q1=<x, y, z, w>. 

However, it is impossible to come up with a right-hand-rule coordinate system that would always perceive a rotation as opposite in direction to the rotation according to another right-handed coordinate system. 

It's strange to me that this conversion was necessary in the first place since the the imu data we were getting in the sim before was coming from the body link directly. If the body link's initial orientation is weird we would need to get the correct orientation by quaternion multiplication ie. q_auv_imu*q_imu_global, , negating parts of the quaternion wouldn't always work to produce the correct orientation. 

In my opinion, we should have separate static link for the imu and fix it in the correct orientation relative to the body_link so we can feed the quaternion through directly - emulating what would occur irl.